buildSrc/settings.gradle

```diff
@@ -0,0 +1,43 @@
+pluginManagement {
+  repositories {
+    google()
+    gradlePluginPortal()
+    mavenCentral()
+  }
+}
+
+dependencyResolutionManagement {
+  versionCatalogs {
+    libs {
+      from(files("../gradle/libs.versions.toml"))
+      if (System.getenv("DEP_OVERRIDES") == "true") {
+        Map<String, String> overrides = new LinkedHashMap()
+        def mergedProperties = System.getenv().plus(System.getProperties())
+        mergedProperties.forEach { key, value ->
+          String keyString = key.toString()
+          if (keyString.startsWith("DEP_OVERRIDE_")) {
+            overrides.put(keyString, value.toString())
+          }
+        }
```

Comment:
Since 7.5 there is `providers.xxxPrefixedBy(...)` for `systemProperties` & `environmentVariables` (and since 8.0 there is the equivalent for `gradleProperties`), might make this entire block and the subsequent code slightly more readable (plus compatible with config cache)

Reply:
Those aren't available in this context